### PR TITLE
Enhancement - Delete member 

### PR DESCRIPTION
**Pull request by yash-chowdhary**
Deleting a member now deletes all the tasks that were assigned to that member.
If an Assignor of a task is deleted, the task still remains.

### CODE REVIEW

File: src/main/java/seedu/club/logic/commands/DeleteCommand.java [hunk 1]
```diff
@@ -47,12 +49,12 @@ public CommandResult executeUndoableCommand() throws CommandException {
         requireToLogIn();
         requireExcoLogIn();
         try {
-            model.deleteMember(memberToDelete);
+            numberOfTasksDeleted = model.deleteMember(memberToDelete);
```
Comment (amrut-prabhu) [inline src/main/java/seedu/club/logic/commands/DeleteCommand.java:52]: I see what you're trying to do here.
But, should `deleteMember` return `numOfTaksDeleted`?

File: src/main/java/seedu/club/model/ClubBook.java [hunk 2]
```diff
@@ -502,6 +508,15 @@ public boolean equals(Object other) {
                 && this.tags.equalsOrderInsensitive(((ClubBook) other).tags));
     }
 
+    @Override
+    public void setLogInMember(Member target) {
+        members.setCurrentlyLogInMember(target);
+    }
+
+    @Override
+    public Member getLogInMember() {
```
Comment (amrut-prabhu) [inline src/main/java/seedu/club/model/ClubBook.java]: Change to `getLoggedInMember`?

**Review by amrut-prabhu — COMMENTED** [review]
Nice work

File: src/main/java/seedu/club/model/ClubBook.java [hunk 3]
```diff
@@ -520,4 +535,21 @@ public void updateTask(Task taskToEdit, Task editedTask) throws DuplicateTaskExc
...
+    /**
+     * Removes all tasks that have been assigned to {@code member}.
+     */
+    public int removeTasksOfMember(Member member) {
+
+        int numberOfTasksRemoved = 0;
+        Iterator<Task> it = tasks.iterator();
+        while (it.hasNext()) {
+            Task task = it.next();
+            if (task.getAssignee().getAssignee().equalsIgnoreCase(member.getMatricNumber().toString())) {
```
Comment (amrut-prabhu) [inline src/main/java/seedu/club/model/ClubBook.java:541]: Idk about this but, maybe you should consider making 
`boolean isTaskAssignedToMember`

File: src/main/java/seedu/club/logic/commands/DeleteCommand.java (hunk 1 above)
Comment (yash-chowdhary) [reply src/main/java/seedu/club/logic/commands/DeleteCommand.java:52]: strictly speaking, it shouldn't. however, there are a lot of implementation overheads especially in terms of testing if i were to use another method.

File: src/main/java/seedu/club/model/ClubBook.java (hunk 3 above)
Comment (yash-chowdhary) [reply src/main/java/seedu/club/model/ClubBook.java:541]: alright. will do 👍

File: src/main/java/seedu/club/model/ClubBook.java (hunk 2 above)
Comment (yash-chowdhary) [reply src/main/java/seedu/club/model/ClubBook.java]: 👍

**Pull request closed without merging**